Userland/Applications/KeyboardSettings/KeyboardSettingsWidget.cpp

```diff
@@ -228,7 +228,13 @@ KeyboardSettingsWidget::KeyboardSettingsWidget()
     m_selected_keymaps_listview->on_selection_change = [&]() {
         auto& selection = m_selected_keymaps_listview->selection();
         m_remove_keymap_button->set_enabled(!selection.is_empty() && keymaps_list_model.keymaps().size() > 1);
-        m_activate_keymap_button->set_enabled(!selection.is_empty());
+        if (selection.is_empty()) {
+            m_activate_keymap_button->set_enabled(!selection.is_empty());
+        } else {
+            auto& highlighted_keymap = keymaps_list_model.keymap_at(selection.first().row());
```

Comment:
Should the button be enabled if the active keymap is not the first selection?

(I don't know either just asking whether you have thought about this)

Reply:
I had not thought about that, thanks for reminding :)

But the thing is, there can be only one active keymap and activating multiple ones doesn't make sense. The activate button should be disabled if there are multiple keymaps selected. Looking at the code, that doesn't seem to be the case either. I'll add a separate commit to the PR doing that

Reply:
Should I add a check there (for disabling the button for multiple selection) anyway?

Reply:
If you cannot select multiple keymaps then this is fine (I guess you could add a VERIFY to ensure the selection has only one item but that's not required)